File: proxy/java/SecurityDoorProxy.java
package io.github.androidhot.proxy.java;

public class SecurityDoorProxy implements Door {
  private SecurityDoor door;

  public SecurityDoorProxy(SecurityDoor door) {
    this.door = door;
  }

  @Override
  public void open() {
    if (checkPassword()) {
      door.open();
    } else {
      System.out.println("The password is wrong, Please check the password.");
    }
  }

  @Override
  public void close() {
    door.close();
  }

  private boolean checkPassword() {
    return false;
  }
}

File: proxy/kotlin/SecurityDoorProxy.kt
package io.github.androidhot.proxy.kotlin

class SecurityDoorProxy(private val door: SecurityDoor) : Door {
  override fun open() {
    if (checkPassword()) {
      door.open()
    } else{
      println("The password is wrong, Please check the password.")
    }
  }

  override fun close() {
    door.close()
  }

  private fun checkPassword(): Boolean {
    return false
  }
}
